### PR TITLE
Change project license to MIT license

### LINKED ISSUE #155: Change license to MIT
Hello @akx and @IcedShake.

Joey and I privately discussed changing the license of this project to MIT and retroactively apply the MIT license to all past releases. Since you contributed code to this repository, you are also rightsholders to this project, I would like to ask you to agree to this license change as well.

Agreeing means that all your code contributions and past and future releases of this project containing your code will be licensed under the MIT license. (For context: exactly that is already happening just with the GPLv3 license. For more information, see below.)

If you have any questions or concerns, please feel free to comment below and we can discuss this.

If you do not agree to this license change or do not respond within a week, we will revert your contributions to move forward with the license change. Of course, we would like to avoid this if possible, so we hope that you agree.

---

TLDR: MIT is more permissive to users of the library. GLPv3 puts restrictions on how/who can use spandrel, which we do not want. We (@joeyballentine and I) consider using GLPv3 license a mistake, and we would like to fix that.

**Motivation:** 
Our goal for spandrel was for the wider community to adopt this library as the primary way to load, save, and run AI upscaling models. Unfortunately, GPLv3 hinders this goal with its strong copyleft. Anyone including spandrel's code in their software distribution would be forced to open source their application. This makes some uses of spandrel (e.g. vendoring) impossible, and spandrel unusable for some people/companies. Therefor, we want to relicense with MIT to lift those restrictions.

**MIT vs GLPv3:** 
MIT and GPLv3 are very similar in most aspects. The main difference is copyleft. As described above, GPLv3 forces some people to open-source their application when using this library. The MIT license has no such restriction. It is more permissive.

**What does changing the license mean?** 
You contributed to this project and therefor licensed your code under the GPLv3 license. This license places certain restrictions on users of spandrel, that we now want to lift. So we are asking you to allow the license of your code to be changed to MIT.

**Do I lose any legal rights to my code?** 
No. You are still the rightsholder of your code contributions. The only thing that change is that people are no longer forced to open source their software for using your code.

**Why not use X license?** 
While there are other permissive licenses, MIT is the most commonly-used open source license. Using less known licenses might hinder the adoption of spandrel, as people might (rightfully) be cautious to use a project with a license that is unknown to them. Not everyone is a lawyer/knowledgeable in software licenses. We should meet our (potential) user where they are if we want them to use our code.

### PR DESCRIPTION
**Pull request by RunDevelopment**
This PR changes the license of the project from GPLv3 to MIT.

This PR will not be merged until all rights holders are informed and agreed to it.

Update: We all agreed to it. Fixes #155.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -15,7 +15,7 @@ keywords = [
     "model arch",
     "model architecture",
 ]
-license = { text = "GPLv3" }
+license = "MIT"
```
Comment (joeyballentine) [inline pyproject.toml:18]: This needs the text field back

Comment (RunDevelopment) [reply pyproject.toml:18]: Why? I looked up examples and they didn't use the `text` field.

Comment (joeyballentine) [reply pyproject.toml:18]: Oh, I guess not. That's what I thought the build error was for, but I guess it just doesnt recognize MIT?

Comment (RunDevelopment) [reply pyproject.toml:18]: Oh, I didn't even look at the errors yet. Hmmm, this might a difference between tooling then. I add the `text` field again.

File: LICENSE [hunk 2]
```diff
@@ -1,674 +1,19 @@
...
-For more information on this, and how to apply and follow the GNU GPL, see
-<https://www.gnu.org/licenses/>.
-
-  The GNU General Public License does not permit incorporating your program
-into proprietary programs.  If your program is a subroutine library, you
-may consider it more useful to permit linking proprietary applications with
-the library.  If this is what you want to do, use the GNU Lesser General
-Public License instead of this License.  But first, please read
-<https://www.gnu.org/licenses/why-not-lgpl.html>.
+Copyright (c) 2024 Joey Ballentine
```
Comment (joeyballentine) [inline LICENSE:1]:
```suggestion
Copyright (c) 2024 The ChaiNNer Organization
```

Does this have to be my name, or can we use our org name for it?

Comment (RunDevelopment) [reply LICENSE]: Sure. I just used your name because you created the repo.

**Review by joeyballentine — APPROVED** [review]
